src/Common/ThreadStatus.cpp

```diff
@@ -60,7 +60,8 @@ struct ThreadStack
     void * getData() const { return data; }
 
 private:
-    static constexpr size_t size = 16 << 10; /// 16 KiB - not too big but enough to handle error.
+    /// 16 KiB - not too big but enough to handle error.
+    static constexpr size_t size = std::max<size_t>(16 << 10, MINSIGSTKSZ);
```

Comment:
Interesting thing about MINSIGSTKSZ under osx is that man page states that ENOMEM will be returned even for size==MINSIGSTKSZ, but the same code works before #25509, so this should be enough.